pkg/querier/queryrange/cache.go

```diff
@@ -0,0 +1,28 @@
+package queryrange
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/cortexproject/cortex/pkg/querier/queryrange"
+)
+
+type CacheKeyLimits interface {
```

Comment:
@cyriltovena Do you think it's excessive to have custom interfaces for this? Should I just inline a `cacheKeyLimits{limits}` when constructing the cache middleware?